File: BI-TJV/semestral/rest-server/src/main/java/wrzecond/controller/TjvAllergenController.java
package wrzecond.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import wrzecond.allergen.TjvAllergenCreateDTO;
import wrzecond.allergen.TjvAllergenReadDTO;
import wrzecond.entity.TjvAllergen;
import wrzecond.service.TjvAllergenService;
import wrzecond.service.TjvEmployeeService;

@RestController
@TjvController.Visibility (
    getByID = TjvController.VisibilitySettings.NONE
)
@RequestMapping("/allergens")
public class TjvAllergenController extends TjvController<TjvAllergen, TjvAllergenReadDTO, TjvAllergenCreateDTO> {

    @Autowired
    public TjvAllergenController (TjvAllergenService allergenService, TjvEmployeeService authService) {
        super (allergenService, authService);
    }

    @Override
    protected String getCreateLocationPath () {
        return "/allergens/%d";
    }

}

File: BI-KOT/semestral/rest-server/src/main/kotlin/wrzecond/controller/TjvAllergenController.kt
package wrzecond.controller

import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import wrzecond.TjvAllergenCreateDTO
import wrzecond.TjvAllergenReadDTO
import wrzecond.TjvAllergenUpdateDTO
import wrzecond.entity.TjvAllergen
import wrzecond.service.TjvAllergenService
import wrzecond.service.TjvEmployeeService

@RestController
@RequestMapping("/allergens")
@TjvController.Visibility (
    getByID = TjvController.VisibilitySettings.NONE
)
class TjvAllergenController (override val service: TjvAllergenService, employeeService: TjvEmployeeService)
: TjvController<TjvAllergen, TjvAllergenReadDTO, TjvAllergenCreateDTO, TjvAllergenUpdateDTO> (service, employeeService) {
    override val createLocationPath: String
        get() = "/allergens/%d"
}